src/pages/ProductDetail.js

```diff
@@ -0,0 +1,132 @@
+
+const useStyles = makeStyles(theme => ({
+  root: {
+    flexGrow: 1,
+    marginTop: '50px',
+    fontFamily: `'Trebuchet MS', 'Lucida Sans Unicode', 'Lucida Grande', 'Lucida Sans', Arial, sans-serif`,
+  },
+  image: {
+    height: '400px',
+    width: '400px',
+    backgroundRepeat: 'no-repeat',
+    backgroundSize: 'contain',
+    [theme.breakpoints.down('sm')]: {
+      height: '300px',
+      width: '280px',
+    },
+  },
+  description: {
+    maxWidth: '400px',
+    marginBottom: 15,
+  },
+  title: {
+    marginBottom: 15,
+  },
+  price: {
+    marginBottom: 15,
+  },
+  rating: {
+    display: 'flex',
+    alignItems: 'center',
+    marginBottom: 15,
+  },
+  star: {
+    color: '#3f51b5',
+    fontSize: '20px',
+    marginLeft: '5px',
+  },
+  size: {
+    display: 'flex',
+    alignItems: 'center',
+    marginBottom: 15,
+  },
+  quantity: {
+    display: 'flex',
+    alignItems: 'center',
+    marginBottom: 30,
+  },
+  addToCart: {
+    fontSize: '22px',
+    marginRight: '8px',
+  },
+  details: {
+    padding: '10px',
+  },
+}));
+
+const ProductDetail = props => {
+  const classes = useStyles();
+  const [productDetails, setProductDetails] = useState({});
+  const { image, description, title, rating, sizes, price } = productDetails;
+  const [currentSize, setCurrentSize] = useState();
+
+  const fetchProductDetails = () => {
+    console.log(props.match.params.id);
+    const details = products.find(({ id }) => id === parseInt(props.match.params.id, 10));
+    setProductDetails(details);
+  };
+
+  useEffect(() => {
+    fetchProductDetails();
+  }, []);
+
+  return productDetails.title ? (
+    <Grid container justify="center" spacing={10} className={classes.root}>
+      <Grid item>
+        <div className={classes.image} style={{ backgroundImage: `url(${image})` }} />
+      </Grid>
+      <Grid item>
+        <div className={classes.details}>
+          <h1 className={classes.title}>{title}</h1>
+          <h3 className={classes.price}>
+            {price && <FormattedNumber value={price} style="currency" currency="INR" />}
+          </h3>
+          <div className={classes.description}>
+            <p>{description}</p>
+          </div>
+          <div className={classes.rating}>
+            <b>{rating}.0</b>
+            <Star className={classes.star} />
+          </div>
+          <div className={classes.size}>
+            <b>Size </b>
+            <SizeContainer
+              value={currentSize}
+              sizes={sizes}
+              onClick={sizeValue => setCurrentSize(sizeValue)}
+            />
+          </div>
+          <div className={classes.quantity}>
+            <b>Quantity </b>
+            <QuantityContainer />
+          </div>
+          <Grid container spacing={8} justify="center">
+            <Grid item>
+              <Button variant="contained" color="primary">
+                BUY NOW
+              </Button>
+            </Grid>
+            <Grid item>
+              {' '}
```

Comment:
Remove this.

Reply:
This a prettier change which comes on running 'npm run format'  you can make changes in the configuration to remove it

Reply:
A detailed explanation for this is, there is an eslint rule to keep one expression per line. 

This means, while coding, there was an extra space left out after the `<Grid item>` tag. and eslint-prettier is trying to preserve that by adding a space in the next line(on the next line because even trailing space is not allowed by one of the eslint rules) as an expression(because there can only be 1 expression per line) 

So by that, this is an unnecessary space, please remove it.

Reply:
Okay